modules/TransitionMixin.js

```diff
@@ -0,0 +1,157 @@
+var warning = require('react/lib/warning');
+var { getParamNames } = require('./PathUtils');
+
+function forEachComponent(components, callback) {
+  if (Array.isArray(components)) {
+    components.forEach(callback);
+  } else if (typeof components === 'object') {
+    for (var key in components)
+      if (components.hasOwnProperty(key))
```

Comment:
At my company we use React-Router inside Salesforce and one of the many quirks of Salesforce is that they modify `Array.prototype` to have a function `remove`. 

Example: 
![salesforce_com_-_developer_edition](https://cloud.githubusercontent.com/assets/3267774/7550629/2db81636-f637-11e4-9c68-4fcb981652ef.png)


Could an `&& components[key] instanceOf React.Component` be added here to facilitate the weirdness?